src/Sarif.Viewer.VisualStudio/CodeAnalysisResultManager.cs

```diff
@@ -637,5 +608,201 @@ internal bool TryResolveFilePathFromSolution(string solutionPath, string pathFro
+                dataCache.RemappedPathPrefixes.Add(new Tuple<string, string>(originalPrefix, resolvedPrefix));
+            }
+
+            return true;
+        }
+
+        // return false means cannot resolve local file and will use embedded file.
+        internal bool VerifyFileWithArtifactHash(SarifErrorListItem sarifErrorListItem, string pathFromLogFile, RunDataCache dataCache, string resolvedPath, string embeddedTempFilePath, out string newResolvedPath)
+        {
+            newResolvedPath = null;
+
+            if (string.IsNullOrEmpty(resolvedPath))
+            {
+                // cannot find corresponding file in local, then use embedded file
+                newResolvedPath = embeddedTempFilePath;
+                return true;
+            }
+
+            if (!dataCache.FileDetails.ContainsKey(pathFromLogFile))
+            {
+                newResolvedPath = resolvedPath;
+                return true;
+            }
+
+            ArtifactDetailsModel fileData = dataCache.FileDetails[pathFromLogFile];
+            string fileHash = this.GetFileHash(this._fileSystem, resolvedPath);
+
+            if (fileHash.Equals(fileData.Sha256Hash))
+            {
+                // found a file in file system which has same hashcode as embeded content.
+                newResolvedPath = resolvedPath;
+                return true;
+            }
+
+            var dialog = new ResolveEmbeddedFileDialog(hasEmbeddedContent: !string.IsNullOrEmpty(embeddedTempFilePath));
+            dialog.ShowModal();
+            if (dialog.Result == ResolveEmbeddedFileDialogResult.None)
+            {
+                // dialog is cancelled.
+                newResolvedPath = null;
+                return false;
+            }
+
+            if (dialog.Result == ResolveEmbeddedFileDialogResult.OpenEmbeddedFileContent)
+            {
+                newResolvedPath = embeddedTempFilePath;
+                return true;
+            }
+
+            if (dialog.Result == ResolveEmbeddedFileDialogResult.OpenLocalFileFromSolution)
+            {
+                newResolvedPath = resolvedPath;
+                return true;
+            }
+
+            if (dialog.Result == ResolveEmbeddedFileDialogResult.BrowseAlternateLocation)
+            {
+                // if returns null means user cancelled the open file dialog.
+                newResolvedPath = this._promptForResolvedPathDelegate(sarifErrorListItem, pathFromLogFile);
+                return !string.IsNullOrEmpty(newResolvedPath);
+            }
+
+            return false;
+        }
+
+        private string GetFileHash(IFileSystem fileSystem, string filePath)
+        {
+            if (!fileSystem.FileExists(filePath))
+            {
+                return null;
+            }
+
+            using (FileStream stream = File.OpenRead(filePath))
+            {
+                return HashHelper.GenerateHash(stream);
+            }
+        }
+
+        private string GetSoltionPath()
```

Comment:
>GetSoltionPath [](start = 23, length = 14)

GetSolutionPatth

Reply:
fixed

---
In reply to: [601284090](https://github.com/microsoft/sarif-visualstudio-extension/pull/384#discussion_r601284090) [](ancestors = 601284090)